src/Compiler/SyntaxTree/SyntaxTree.fsi

```diff
@@ -354,9 +354,26 @@ type SeqExprOnly =
     /// Indicates if a for loop is 'for x in e1 -> e2', only valid in sequence expressions
     | SeqExprOnly of bool
 
-/// Represents the location of the separator block + optional position
-/// of the semicolon (used for tooling support)
-type BlockSeparator = range * pos option
+/// Represents the location of the separator block and optional position of the semicolon (used for tooling support)
+[<NoEquality; NoComparison; RequireQualifiedAccess>]
+type BlockSeparator =
+    /// A separator consisting of a semicolon ';'
+    /// range is the range of the semicolon
+    /// position is the position of the semicolon (if available)
+    | Semicolon of range: range * position: pos option
+    /// A separator consisting of a comma ','
+    /// range is the range of the comma
+    /// position is the position of the comma (if available)
+    | Comma of range: range * position: pos option
+
+    // A separator consisting of a newline
+    /// range is the range of the newline
+    /// position is the position of the newline (if available)
+    | Offside of range: range * position: pos option
```

Comment:
Why do we need to capture this case? And especially the new line? Are there any cases where we rely on it? If yes, let's review them, please. It seems very suspicious. It effectively captures implementation details of how things are parsed and we should not rely on it in any way.

Reply:
Why would we ever want to model it? If that's an implicit separator created by the unrelated subsequent tokens, we should not be relying on it in any way.